### ISSUE TITLE
Umbrella throwing error: distutils module not found

### ISSUE THREAD

**Issue by mombasa123** [umbrellaplug/umbrellaplug.github.io#38]
Hi,

Tried to move to Umbrella from Venom (hacked version) but getting below error on Discover Movies->any menu item.   Happens  on Search (immediately), TV Shows etc.

Initially tried the install on an existing Kodi and got above. Then tried a fresh new Kodi Matrix build on OSMC Verio 4k+ and same.

_2022-07-31 20:02:44.721 T:2814    ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'ModuleNotFoundError'>
                                                   Error Contents: No module named 'distutils.log'
                                                   Traceback (most recent call last):
                                                     File "/home/osmc/.kodi/addons/plugin.video.umbrella/umbrella.py", line 18, in <module>
                                                       router.router(url)
                                                     File "/home/osmc/.kodi/addons/plugin.video.umbrella/resources/lib/modules/router.py", line 49, in router
                                                       from resources.lib.menus import movies
                                                     File "/home/osmc/.kodi/addons/plugin.video.umbrella/resources/lib/menus/movies.py", line 21, in <module>
                                                       from resources.lib.modules import mdblist
                                                     File "/home/osmc/.kodi/addons/plugin.video.umbrella/resources/lib/modules/mdblist.py", line 8, in <module>
                                                       from distutils.log import error
                                                   ModuleNotFoundError: No module named 'distutils.log'
                                                   -->End of Python script error report<--_

I googled distutils but its deprecated for python 3 and Matrix so not sure how Umbrella is using? That's the extent of my knowledge sadly.

Umbrella (debug) and kodi logs uploaded:

Umbrella: https://paste.kodi.tv/ugijucalic
Kodi: https://paste.kodi.tv/kiwogonawa

Thanks & regards

EDIT; Just tried installing /  running on Windows and all good so looks like an OSMC issue but confirmation would be appreciated so I can take it to OSMC support. Cheers

**Comment by umbrellaplug** [comment]
distutils is a built in python module required by the plugin. This looks like you're having an issue with your kodi build and python (the language used to program addons). I'd check what version of python they are using in this build or if there's another build they can offer you of osmc. This has to do with the version of python they've used.

**Comment by mombasa123** [comment]
Thanks - am raising to Sam and the OSMC guys.

**Comment by zjoasan** [comment]
As I said on OSMC-forum, "sudo apt-get install python3-distutils", but as I also said on the forum the python install tool pip is another gold nugget to have in your system.

**Comment by mombasa123** [comment]
> As I said on OSMC-forum, "sudo apt-get install python3-distutils", but as I also said on the forum the python install tool pip is another gold nugget to have in your system.

Cheers, much appreciated. Will give it a spin.

**Comment by umbrellaplug** [comment]
Closing this as it appears unrelated the the plugin. Good luck. I hope you get it figured out.

**Comment by graham8** [comment]
@umbrellaplug It seems distutils will be removed in Python 3.10.  You should consider using setuptools.

**Comment by umbrellaplug** [comment]
Thanks for the heads ups.
I would like to be honest here because I feel that's important. This issue was my fault. I have since corrected it and I should have come back here and commented on this issue to explain so I apologize for that. It has been a busy week, but no excuses. 

The issue was caused by using visual studio with some python extensions installed to make coding a little easier. I type my own code out and as I was typing I must have caused visual studio to think I was trying to import distutils for something. I never intended to import distutils into that module nor did I need to but somehow it slipped past me. I have double checked that module since and removed and unused imports caused by this and switched to a different tool for coding since. 

Thanks for reporting this and for coming back and commenting. Hopefully this fully explains everything and all should be working as expected now.
